core/src/test/java/com/linecorp/armeria/server/AnnotatedHttpServiceTest.java

```diff
@@ -482,7 +533,7 @@ public void testAnnotatedHttpService() throws Exception {
             }
             // Run case 6 but with not-mapped HTTP method (Post).
             try (CloseableHttpResponse res = hc.execute(new HttpPost(rule.httpUri("/2/string/blah")))) {
-                assertThat(res.getStatusLine().toString(), is("HTTP/1.1 405 Method Not Allowed"));
+                assertThat(res.getStatusLine().toString(), is("HTTP/1.1 404 Not Found"));
```

Comment:
I still think we should respond with `405 Method Not Allowed` because there *is* a resource at the specified path but only a certain set of methods are accepted. #691 may be useful for implementing this behavior.